developers/blobstream-rollups.md

```diff
@@ -0,0 +1,123 @@
+
+Note: **Generating/verifying share commitment proofs is still not supported.
+It still needs tooling to generate the proofs on the node side,
+and verifying them on the Solidity side which will be built in the upcoming months.
+Thus, the [sequence of spans](#optimistic-rollups-that-uses-a-sequence-of-spans) approach is preferable at the moment.
+Also, the gas cost of verifying all the above proofs needs to be investigated.**
+
+#### Protobuf deserialization contract
+
+Another alternative when using the share commitment to reference the rollup data is to parse the PFB,
+i.e., the protobuf serialized PayForBlob transaction,
+inside the rollup settlement contract and verify that the parsed share commitment is the same.
+Then, proving the inclusion of the PFB to the data root tuple root.
+More details on this can be found in [ADR-011](https://github.com/celestiaorg/celestia-app/blob/main/docs/architecture/adr-011-optimistic-blob-size-independent-inclusion-proofs-and-pfb-fraud-proofs.md). 
+
+The issue with this approach is needing a library to parse the protobuf in solidity and the gas cost that comes with it.
+Otherwise, proving the PFB to the data root tuple root is cheaper than proving the blob,
+because it needs only proving two shares.
+
+## Zk-Rollups
+
+Zk-rollups, aka validity rollups, can also use Celestia as a DA and Blobstream to verify that the data was posted.
+However, the submission process is different from the above constructions,
+since there are no fraud proofs, and everything should be verified when submitting the headers.
+
+So, when posting to the settlement contract,
+the rollup data in Celestia can be referenced either using a `share commitment` or a `sequence of spans`,
+and a `height`, similar to the previous constructions.
+
+### Zk-rollups that uses a sequence of spans
+
+So, when submitting the headers to the rollup settlement contract,
+the settlement contract will need to verify the following:
+
+- Zk-proof of the state transitions, as traditional zk-rollups do.
+- Verify that the `sequence of spans` is [valid](https://github.com/celestiaorg/blobstream-contracts/blob/master/docs/inclusion-proofs.md),
+  i.e., is part of the Celestia block referenced by its height,
+  as described in the previous section.
+- Zk-proof of the rollup data to the data root.
+  The verification process of this should accept a commitment as input
+  so that the contract makes sure it's the correct value that's being saved.
+  The commitment can be the data root and the sequence of spans.
+  And, when the rollup data is proven inside the circuit to the data root,
+  the used data root is asserted to be the input one.
+  Similarly, the data's location is asserted to be the same as the input sequence of spans.
+  While these same arguments are the ones used as part of the `sequence of spans` verification above.
+
+Once these are valid, the settlement contract can be sure that the state transitions are valid,
```

Comment:
in general, I think this post tip toes into rollup proving systems, but imo, this actually distracts from the main points that we are trying to get at.